templates/aws-hosted-cp/templates/awscluster.yaml

```diff
@@ -4,6 +4,8 @@ metadata:
   name: {{ include "cluster.name" . }}
   annotations:
     cluster.x-k8s.io/managed-by: k0smotron
+  finalizers:
+  - hmc.mirantis.com/do-not-delete
```

Comment:
Shouldn't we add it to `aws-standalone-cp` as well?

Reply:
I don't think so. The issue was with AWSCluster removed before underlying objects. Or do you see any other cases?

Reply:
It's especially important for the hosted template. please add and verify if anything else is needed.

Reply:
@slysunkin @Kshatrix This probably won't work. In my tests on initial apply the `finalizers` field was emptied (thus the need for patch in the docs). This should be checked thoroughly. 
`AzureCluster` is also affected, so we should add fix for it as well,

Reply:
I'm working on Azure now